returnn/tf/layers/basic.py

```diff
@@ -908,12 +917,17 @@ def get_out_data_from_opts(cls, name, sources=(), start=None, size=None, **kwarg
     :param int|None size:
     :rtype: Data
     """
-    from ..util.data import DimensionTag
-    input_data = get_concat_sources_data_template(sources).copy_as_batch_spatial_major()
-    if start:
-      input_data.beam = SearchBeam.get_combined_beam(input_data.beam, start.output.beam)
-    new_dim_tag = DimensionTag(kind=DimensionTag.Types.Spatial, description="%s:slice_nd" % name, dimension=size)
-    return input_data.copy_template_replace_dim_tag(axis=1, new_dim_tag=new_dim_tag, name="%s_output" % name)
+    start = start.output.copy_as_batch_major()
+    indices_data = start.copy_template(name="%s_gather_indices" % name)
+    indices_data = indices_data.copy_add_spatial_dim(spatial_dim_axis=start.batch_ndim, dim=size,
+                                                     auto_time_dim_axis=False)
+    position = InternalLayer(network=sources[0].network, name="%s_internal" % indices_data.name, output=indices_data)
+    return GatherLayer.get_out_data_from_opts(
+      name="%s_gather" % name,
+      sources=sources,
+      position=position,
+      axis="T"
+    )
```

Comment:
Formatting: the `)` should come in the previous line, not in its own line.

Logically: do not use `T` here. That's too ambiguous. Use the exact dim tag.